### PR TITLE
Fix `muxer.Attach`

### PR DESCRIPTION
**Pull request by iwahbe**
The muxer's attach method is broken when setup with a nil host. It panics with a nil pointer exception. This replaces the previous implementation with a straight dispatch call to subsidiary servers.

### CODE REVIEW

File: x/muxer/muxer.go [hunk 1]
```diff
@@ -394,19 +395,16 @@ func (m *muxer) GetPluginInfo(ctx context.Context, e *emptypb.Empty) (*rpc.Plugi
...
+	for i, s := range m.servers {
+		s := s
+		attach[i] = func() error {
+			_, err := s.Attach(ctx, req)
+			return err
 		}
 	}
-	// Here we override the underlying host. This should replace the host instance of
-	// each subsidiary provider.
-	*m.host = *host
```
Comment (mjeffryes) [inline x/muxer/muxer.go:408]: does it matter that we're no longer setting `m.host`? Do we still need that field?

**Review by mjeffryes — COMMENTED** [review]
So it looks like before this change we were not muxing the Attach call, but now we are? The new behavior and tests look good, but I don't think I have full context to determine if the `NewHostClient` bits are safe to remove. (I did note that the one other place the `host` field is uses is in `WarnF` around line 513.)

File: x/muxer/muxer.go (hunk 1 above)
Comment (iwahbe) [reply x/muxer/muxer.go:408]: Good catch. I believe that it does matter at the margin. Fixed in the next commit.

File: x/muxer/muxer.go [hunk 2]
```diff
@@ -394,19 +395,18 @@ func (m *muxer) GetPluginInfo(ctx context.Context, e *emptypb.Empty) (*rpc.Plugi
 }
 
 func (m *muxer) Attach(ctx context.Context, req *rpc.PluginAttach) (*emptypb.Empty, error) {
-	host, err := provider.NewHostClient(req.GetAddress())
-	if err != nil {
-		return nil, err
-	}
-	if m.host != nil {
-		if err := m.host.Close(); err != nil {
-			return nil, err
```
Comment (mjeffryes) [inline x/muxer/muxer.go:403]: Do we need to close the old host if it exists?

**Review by mjeffryes — APPROVED** [review]

**Pull request merged**